### PR TITLE
Fix data migrations

### PR DESCRIPTION
**Pull request by LefterisJP**
Fixes a number of issues with data migrations.

- A failed data migration should be visible to the user.
- Data migration 1 does not depend on state but reads from DB
Previously it was trying to manipulate the DB while reading the state
of the app which was assuming:
    1. That exchanges manager was initialized at the point the migration
    is called. Bad assumption
    2. There is no exchange query ranges in the DB that we have no
    credentials for. This can also happen.
- Added a mock to not do any migrations during unlock. Doing a
  migration during unlock killed the test and made it unreliable.
- The test is fixed and simplified.
- The migration DB for the test now also contains balancer_trades
  example to make sure our logic handles it fine
- check no errors/warnings happened during migration since we now also
  log an error properly via the msg aggregator
- Failed data migration must block more migrations and not update DB. Also add test for it.

### CODE REVIEW

File: rotkehlchen/tests/data_migrations/test_migrations.py [hunk 1]
```diff
@@ -1,29 +1,19 @@
...
-    Timestamp,
-    TradeType,
-)
+from rotkehlchen.typing import AssetAmount, Fee, Location, Price, Timestamp, TradeType
 
 
 @pytest.mark.parametrize('use_custom_database', ['data_migration_v0.db'])
-@pytest.mark.parametrize('data_migration_version', [0])
-def test_migration_1(
-    use_custom_database,  # pylint: disable=unused-argument
```
Comment (LefterisJP) [inline rotkehlchen/tests/data_migrations/test_migrations.py:21]: @yabirgb you do not need to be adding the fixtures in the test again if they are already pulled by a fixture you have added. In this case the `rotkehlchen_api_server`

File: rotkehlchen/tests/utils/database.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import os
+import random
```
Comment (LefterisJP) [inline rotkehlchen/tests/utils/database.py:2]: @yabirgb  Make sure your editor uses isort and same version as pinned in requirements. This is my editor using isort and correcting location.

File: rotkehlchen/data_migrations/manager.py [hunk 3]
```diff
@@ -41,10 +43,14 @@ def maybe_migrate_data(self) -> None:
...
 
-    def _perform_migration(self, migration: MigrationRecord) -> None:
+    def _perform_migration(self, migration: MigrationRecord) -> bool:
+        """Performs a single data migration and returns boolean for success/failure"""
         try:
             kwargs = migration.kwargs if migration.kwargs is not None else {}
             migration.function(rotki=self.rotki, **kwargs)
         except BaseException as e:  # lgtm[py/catch-base-exception]
-            error = f'Failed to run soft migration from version {migration.version} : {str(e)}'
-            log.error(error)
```
Comment (LefterisJP) [inline rotkehlchen/data_migrations/manager.py:50]: @yabirgb  with this user has absolutely no indicaton that something happened. In fact ... I spent about 20 mins trying to figure out why the heck no update was happening only to realize there is no feedback. Just a log entry.

File: rotkehlchen/data_migrations/manager.py [hunk 4]
```diff
@@ -32,7 +32,9 @@ def maybe_migrate_data(self) -> None:
         current_migration = settings.last_data_migration
         for migration in MIGRATION_LIST:
             if current_migration < migration.version:
-                self._perform_migration(migration)
+                if self._perform_migration(migration) is False:
```
Comment (LefterisJP) [inline rotkehlchen/data_migrations/manager.py:35]: @yabirgb If a migration fails we STOP. Before we were having failed migrations and the code was happily continuing onto the next one and also writing to the DB that migration was complete.

File: rotkehlchen/tests/data_migrations/test_migrations.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/tests/data_migrations/test_migrations.py:21]: With the fixtures I had a lot of trouble. Thanks

**Comment by LefterisJP** [comment]
All green. So beautiful.

**Pull request merged**